hr_payroll_account_operating_unit/models/hr_payslip.py

```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-17 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# Copyright 2016-17 Serpent Consulting Services Pvt. Ltd.
+#   (<http://www.serpentcs.com>)
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+
+from odoo.tools.translate import _
+from odoo import api, models
+from odoo.exceptions import UserError
+
+
+class HrPayslip(models.Model):
+
+    _inherit = 'hr.payslip'
+
+    @api.multi
+    def write(self, vals):
+        res = super(HrPayslip, self).write(vals)
+        if 'move_id' in vals and vals['move_id']:
+            for slip in self:
+                if slip.contract_id and slip.contract_id.operating_unit_id:
+                    slip.move_id.write({'operating_unit_id':
+                                        slip.contract_id.operating_unit_id.id})
+                    if slip.move_id.line_ids:
+                        slip.move_id.line_ids.\
+                            write({'operating_unit_id':
+                                   slip.contract_id.operating_unit_id.id})
+        return res
+
+    @api.multi
+    def action_payslip_done(self):
+        OU = None
+        for slip in self:
+            # Check that all slips are related to contracts
+            # that belong to the same OU.
+            if OU:
```

Comment:
I was wondering if there is a way to map the contracts by different OU and if there was more than one raise the error. But I'm not really sure and this works. It's ok to me.